rskj-core/src/main/java/co/rsk/net/light/LightProcessor.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2020 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.net.light;
+
+import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
+import co.rsk.db.RepositorySnapshot;
+import co.rsk.db.RepositoryLocator;
+import co.rsk.net.light.message.*;
+import org.ethereum.core.*;
+import org.ethereum.net.message.Message;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.db.BlockStore;
+import org.ethereum.db.TransactionInfo;
+import org.ethereum.vm.DataWord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.LinkedList;
+import java.util.List;
+
+
+/**
+ * Created by Julian Len and Sebastian Sicardi on 21/10/19.
+ */
+public class LightProcessor {
+    private static final Logger logger = LoggerFactory.getLogger("lightprocessor");
+    // keep tabs on which nodes know which blocks.
+    private final BlockStore blockStore;
+    private final RepositoryLocator repositoryLocator;
+    private final Blockchain blockchain;
+
+    public LightProcessor(@Nonnull final Blockchain blockchain,
+                          @Nonnull final BlockStore blockStore,
+                          @Nonnull final RepositoryLocator repositoryLocator) {
+        this.blockchain = blockchain;
+        this.blockStore = blockStore;
+        this.repositoryLocator = repositoryLocator;
+    }
+    /**
+     * processBlockReceiptsRequest sends the requested block receipts if it is available.
+     * @param requestId the id of the request
+     * @param blockHash   the requested block hash.
+     * @param lightPeer
+     */
+    public void processGetBlockReceiptsMessage(long requestId, byte[] blockHash, LightPeer lightPeer) {
+        String blockHashLog = Hex.toHexString(blockHash);
+        logger.trace("Processing block receipts request {} block {}", requestId, blockHashLog);
+        final Block block = blockStore.getBlockByHash(blockHash);
+
+        if (block == null) {
+            // Don't waste time sending an empty response.
+            return;
+        }
+
+        List<TransactionReceipt> receipts = new LinkedList<>();
+
+        for (Transaction tx :  block.getTransactionsList()) {
+            TransactionInfo txInfo = blockchain.getTransactionInfo(tx.getHash().getBytes());
```

Comment:
Could this possibly return null?

Reply:
This is weird, all transactions return a receipt, it doesn't matter if the transaction was successful or not.